mdata/cache/ccache_metric_test.go

```diff
@@ -203,6 +205,73 @@ func TestCorruptionCase1(t *testing.T) {
 	})
 }
 
+func getRandomNumber(min, max int) int {
+	return rand.Intn(max-min) + min
+}
+
+// getRandomRange returns a range start-end so that
+// end >= start and both numbers drawn from [min, max)
+func getRandomRange(min, max int) (int, int) {
+	start := getRandomNumber(min, max)
+	end := getRandomNumber(start, max)
```

Comment:
actually i just realized that this randomness is skewed to the end of the range because, assuming the distribution of `start` across `min - max` is even, and considering that `end` always must be `>= start`, then the middle of the resulting range returned from `getRandomRange()` at average will above the middle of the range `min-max`.
that's not really a problem, just a note

Reply:
we could keep a counter, and every other time, draw an `end` first

Reply:
true, we could also just get two random numbers and return the lower one as `start` and the higher as `end`

Reply:
oh that's elegant I like that.